schemas/string_interpolation/definitions.json

```diff
@@ -0,0 +1,143 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema",
+  "string_with_placeholders": {
+    "oneOf": [
+      {
+        "description": "String with no placeholders.",
+        "type": "string"
+      },
+      {
+        "description": "An object that represents a string with placeholders.",
+        "type": "object",
+        "properties": {
+          "text": {
+            "type": "string"
+          },
+          "placeholders": {
+            "$ref": "#/placeholders"
+          },
+          "additionalProperties": false,
+          "required": [
+            "text",
+            "placeholders"
+          ]
+        }
+      }
+    ]
+  },
+  "placeholders": {
+    "description": "A mapping of string placeholders to their values.",
+    "type": "array", 
+    "items": {
+      "$ref": "#/placeholder"
+    },
+    "minItems": 1
+  },
+  "placeholder": {
+    "description": "A mapping of a string placeholder to it's value. The value can also be transformed.",
+    "type": "object",
+    "properties": {
+      "placeholder": {
+        "description": "The placeholder in the string that will be replaced e.g. 'test' is the placeholder in 'My {test} string'.",
+        "type": "string"
```

Comment:
`placeholder` within `placeholders` should be unique. Don't see a reason for it to not be?.

Reply:
Yes, I'll look at adding some validation for that. I'll also look at adding validation to check that the placeholders in the source string are defined in the placeholders definition.